### PR TITLE
Add MockTriggerTime setting to return events and spot interruptions at a specific time

### PR DESCRIPTION
**Pull request by bruecktech**
*Description of changes:*

This adds a new config setting to specify an RFC3339 timestamp as an alternative to a mock delay. When the timestamp is reached the server will return the mock events or spot interruption.
We use this feature to simulate the interruption of many spot instances at once which requires a better timing than the startup delay

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cmd/root/root.go [hunk 1]
```diff
@@ -77,6 +78,7 @@ func NewCmd() *cobra.Command {
 	cmd.PersistentFlags().StringP(gf.ConfigFileFlag, "c", "", "config file for cli input parameters in json format (default: "+cfg.GetDefaultCfgFileName()+")")
 	cmd.PersistentFlags().BoolP(gf.SaveConfigToFileFlag, "s", false, "whether to save processed config from all input sources in "+cfg.GetSavedCfgFileName()+" in $HOME or working dir, if homedir is not found (default: false)")
 	cmd.PersistentFlags().Int64P(gf.MockDelayInSecFlag, "d", 0, "mock delay in seconds, relative to the application start time (default: 0 seconds)")
+	cmd.PersistentFlags().String(gf.MockTriggerTimeFlag, "", "mock trigger time in RFC3339 (default: none)")
```
Comment (brycahta) [inline pkg/cmd/root/root.go:81]: can you note that `MockTriggerTime` will take priority over `MockDelayInSec`

File: test/e2e/testdata/aemm-config-integ.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
   },
   "spot": {
     "action": "terminate",
-    "time": "2020-01-07T01:03:47Z"
+    "time": "2020-01-07T01:03:47Z",
+    "trigger-time": "2020-01-07T01:01:47Z"
```
Comment (brycahta) [inline test/e2e/testdata/aemm-config-integ.json:14]: this shouldn't be needed here. If anything you could update this file with `mock-trigger-time` on the **root** level and consume this value in the e2e test. However, if you choose to pass in the value as a flag instead, then no change needed here.

**Comment by bruecktech** [comment]
@brycahta thanks for the feedback. I incorporated it in the latest commit. Please have a look.

File: README.md [hunk 3]
```diff
@@ -245,25 +247,26 @@ Usage:
   ec2-metadata-mock spot [--action ACTION] [flags]
 
 Aliases:
-  spot, spotitn
+  spot, spot
```
Comment (brycahta) [inline README.md:250]: this should stay as `spotitn` since it's a valid alias

File: test/e2e/cmd/root-test [hunk 4]
```diff
@@ -87,4 +119,15 @@ $start_cmd &
 ROOT_PID=$!
 test_root_delay $ROOT_PID $DELAY_IN_SEC
 
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  mock_trigger_time=$(date --date "${DELAY_IN_SEC} seconds" +%Y-%m-%dT%T%:z)
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  mock_trigger_time=$(date -v+${DELAY_IN_SEC}S +%Y-%m-%dT%T%z | sed 's@^.\{22\}@&:@')
+fi
+echo "foo: $mock_trigger_time"
```
Comment (brycahta) [inline test/e2e/cmd/root-test:127]: thanks for adding this check! Could you update the 'foo' to mock-trigger-time or remove the echo completely?

File: README.md (hunk 3 above)
Comment (bruecktech) [reply README.md:250]: I think this was changed in this [commit](https://github.com/aws/amazon-ec2-metadata-mock/commit/7621e835590920881c85f81467f3df0df8ab7513#diff-9c3028e9a33c2fdb57e5fb8c9ced84dfL70) already. I just copy-pasted the help output for updating the readme

File: test/e2e/cmd/root-test (hunk 4 above)
Comment (bruecktech) [reply test/e2e/cmd/root-test]: seems it was too late already when i wrote this 😄 
removed it now

File: README.md (hunk 3 above)
Comment (brycahta) [reply README.md:250]: ah, got it I'll look into it. I'll approve+merge once the travis checks pass!

**Pull request merged**